codegen/core/src/main/scala/kalix/codegen/ModelBuilder.scala

```diff
@@ -52,36 +55,37 @@ object ModelBuilder {
     def fromService(service: Service): Model =
       Model.empty.addService(service)
 
-    def fromEntity(entity: Entity): Model =
+    def fromEntity(entity: StatefulComponent): Model =
       Model.empty.addEntity(entity)
   }
 
   /**
    * The Kalix service definitions and entities that could be extracted from a protobuf descriptor
    */
-  case class Model(services: Map[String, Service], entities: Map[String, Entity]) {
-    def lookupEntity(service: EntityService): Entity = {
-      entities.getOrElse(
+  case class Model(services: Map[String, Service], statefulComponents: Map[String, StatefulComponent]) {
+    def lookupEntity(service: EntityService): StatefulComponent = {
+      statefulComponents.getOrElse(
         service.componentFullName,
         throw new IllegalArgumentException(
-          "Service [" + service.messageType.fullyQualifiedProtoName + "] refers to entity [" + service.componentFullName +
-          s"], but no entity configuration is found for that component name. Entities: [${entities.keySet.mkString(", ")}]"))
+          "Service [" + service.messageType.fullyQualifiedProtoName + "] refers to stateful [" + service.componentFullName +
```

Comment:
`refers to stateful ["abc"]` vs `refers to component ["abc"]` or `refers to stateful component ["abc"]`?

Also, why using both concatenation (`+`) and string interpolation (`s`) together?

Reply:
Fixed, I think that mixing was used to avoid very long line.